### PR TITLE
[chore] [receiver/awscloudwatch] Use comparetest.CompareLogs in tests

### PR DESCRIPTION
**Pull request by dmitryax**
Use comparetest.CompareLogs in tests instead of the same local functions.

### CODE REVIEW

**Review by codeboten — APPROVED** [review]
Nice!

File: receiver/awscloudwatchreceiver/testdata/processed/prefixed.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
                     "logRecords": [
                         {
                             "timeUnixNano": "1665166251014000000",
-                            "observedTimeUnixNano": "1665166998995360000",
+                            "observedTimeUnixNano": "1673025345210224000",
```
Comment (codeboten) [inline receiver/awscloudwatchreceiver/testdata/processed/prefixed.json:32]: Is this change necessary if observed timestamp is ignored?

**Review by djaglowski — APPROVED** [review]

File: receiver/awscloudwatchreceiver/testdata/processed/prefixed.json (hunk 1 above)
Comment (dmitryax) [reply receiver/awscloudwatchreceiver/testdata/processed/prefixed.json:32]: It's not. Thanks for catching
